### PR TITLE
refactor: create grouped data structure for message list [WPB-6282]

### PR DESCRIPTION
**Pull request by atomrc**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6282" title="WPB-6282" target="_blank">TaskWPB-6282</a>  [web] have a grouped data structure for message grouping
  </td></table>
  <br />
 

## Description

In order to have better control of how we render messages in the message list and in order to ease grouping strategies, we want to have a grouped data structure (rather than just a flat list of messages). 

So instead of `Message[]` data type to render messages, we now have a `(Marker|MessageGroup)[]` data type. 

`Marker` represents a time marker (messages from today, or from a particular hour or unread)
`MessageGroup` represents messages  from the same sender that should be grouped together because sent roughly at the same time

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/style/content/conversation/message-list.less [hunk 1]
```diff
@@ -244,55 +244,6 @@
   height: 16px;
 }
 
-// MESSAGE - TIMESTAMP
-.message-timestamp {
```
Comment (atomrc) [inline src/style/content/conversation/message-list.less:248]: this has been migrated to the `Marker.styles.ts` file

File: src/script/components/MessagesList/Message/types.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+export type ScrollToElement = (elm: {center?: boolean; element: HTMLElement}, isUnread?: boolean) => void;
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/types.ts:20]: EOL missing

Comment (atomrc) [reply src/script/components/MessagesList/Message/types.ts:20]: It's just Github not showing it, but it's there 
image

**Review by przemvs — APPROVED** [review]
🚀

**Pull request merged**